### DISCUSSION TITLE
What sort of venue data should be easily exportable?

### DISCUSSION THREAD

**Discussion by evolve2k** [koalagator/koalagator#18] [General]
This discussion relates to the interface features and data available on /venues. 
See https://calagator.org/venues to demo how this currently is pre any change (correct as at June 2024).

![SCR-20240622-ntet](https://github.com/koalagator/koalagator/assets/114459/b45d993a-890b-4529-a00f-4fef39d56784)

The current feature set (carried forward from Calagators 2010's design decisions), is that the venue data is easily exportable, essentially public. I'd expect this reflects values of maximum openness from the original project creators.

In 2024 though I suspect the risk of this function being used by bad/commercial actors over users wanting the data to extend the public good/commons is very likley. Eg I suspect that the list will more than likley be exploited.

My inclination is to adjust the feature set to the following:

Public unathenticated 

CAN:
- See Most Active venues list
- See Newest venue list
- Search for a venue
- View Big Map

CANNOT (Restrict)
- List all venues (risk of unmanaged data extraction for spammers)
- Export venue list (KML, JSON)

--- 

I'd propose the following, 
That "List all venues" and Export Venue List, be restricted to Admin users only at this stage.

On the basis that in 2024 as we gather phone numbers, email addresses and other valuable venue data that most users are not needing to download a listing, and for any legitimate community use cases, having Admins be able to perform this function, they can share this data on a request/need basis with other legitimate community users who are seeking to use this data for good collective benefit (eg not spammers/unethical site cloners).

Further, if a spammer, bad/actor really wants to harvest the venue data for ill-will, let them put in the work to setup a custom scraper and mess around. Raise a separate discussion/issue if you think its important to detect/protect against scraping like this.

--

**Comment by evolve2k** [comment]
@danascheider what do you think?

**Comment by oscaralanpierce** [comment]
@evolve2k The one thing I'm wondering is how this data could be harmful if it fell into the wrong hands. I don't necessarily see a need for public, unauthenticated users to be able to download JSON data but I also am not really sure why they shouldn't be able to. Can you expand on what risks you see?

To be clear, I'm not opposed to these access rules. I'm just not sure they are needed.

Also, sorry for the slow responses - I'm still settling into my new place and also don't have Internet access here yet so I might be a little harder to reach than usual for a few days.

**Comment by evolve2k** [comment]
I'd say the main one is spammers. This is an issue on the current calagator.com site. It's not so much private data, more that companies email addresses and phone numbers end up in spam lists. While there's some risk of that anyway, I think the easy access increases the risk.  Right now it's as quick as export all then email all.

If we have venues getting spammed from being on our site, we risk venues asking to be remove from our site, affecting user functionality. This is pre-emptivly improving how we steward venues data.

You agree we can remove the unathenticated data download at this stage?
(Folks can always reopen this when a real use case arisis)

_Ps. No stress on delay, didnt even notice, wishing you well in your new place._

**Reply by oscaralanpierce** [reply]
Yes, I think that's reasonable. Thanks!
